packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-preview.tsx

```diff
@@ -167,12 +190,20 @@ export const PipelinePreview: React.FunctionComponent<PipelinePreviewProps> = ({
 const mapState = (state: RootState) => {
   const pipeline = getPipelineStageOperatorsFromBuilderState(state);
   const lastStage = pipeline[pipeline.length - 1] ?? '';
-  const { isLoading, previewDocs } = state.pipelineBuilder.textEditor.pipeline;
+  const { isLoading, previewDocs, serverError, stageOperators } =
+    state.pipelineBuilder.textEditor.pipeline;
+  const isMissingAtlasSupport = isMissingAtlasStageSupport(
+    state.env,
+    serverError
+  );
+  const atlasOperator = getAtlasOnlyOperators(stageOperators)[0] ?? '';
```

Comment:
Is it a safe assumption that the first atlas-only stage is the one that cause the error? Maybe there is a way to get this from error somehow? 🤔